packages/hash/frontend/src/blocks/page/component-view.tsx

```diff
@@ -195,21 +191,16 @@ export class ComponentView implements NodeView {
               }}
             >
               <BlockLoader
-                key={entityId} // reset the component state when the entity changes, e.g. to reset the data map state
-                blockEntityId={entityId}
+                key={entityId} // reset the component state when the entity changes
+                blockEntityId={
+                  childEntity?.metadata.editionId.baseId as EntityId | undefined
+                } // @todo make this always defined
+                blockEntityTypeId={this.block.meta.schema as VersionedUri} // @todo-0.3 remove when @blockprotocol/core types updated
```

Comment:
Is this tracked in an asana task?

Reply:
No – I'm just going to come back and fix all these `todo-0.3` when we have consistency between the BP types and the HASH types (or least a strategy to reconcile them without assertions)

Reply:
That could work (although a bit of a bus factor there 😅)

Just another possible approach, in the backend we've started putting asana task links in almost all the TODOs. If we had a task to handle that reconciliation, we could just put the asana task in all of these and `ctrl+f` for that, it'd make it easier for someone in the future to also be able to guess that something was missed, rather than needing to know what `todo-0.3` is.

Not a big thing though

Reply:
Created a [task](https://app.asana.com/0/1203543021352041/1203756383156843/f) (internal) in which I'll do this. Since I'll be fixing them this week I don't think we need to link the task in all these todos, although I agree with the principle as a good thing to do to make sure context isn't lost

If I don't fix some of the todos I'll create tasks and link them